libimage/runtime.go

```diff
@@ -168,6 +168,11 @@ func (r *Runtime) LookupImage(name string, options *LookupImageOptions) (*Image,
 		options = &LookupImageOptions{}
 	}
 
+	// Docker compat: strip off the tag iff name is tagged and digested
+	// (e.g., fedora:latest@sha256...).  In that case, the tag is stripped
+	// off and entirely ignored.  The digest is the sole source of truth.
+	name = normalizeTaggedDigestedString(name)
+
```

Comment:
Same here, the code should first decide what the string means, and only normalize the relevant cases.